packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -679,6 +679,14 @@ export class SnippetString {
         return this;
     }
 
+    appendChoice(values: string[], number: number = this._tabstop++): SnippetString {
+        const value = values.map(s => s.replace(/\$|}|\\|,/g, '\\$&')).join(',');
+
+        this.value += `\$\{${number}|${value}|\}`;
+
+        return this;
+    }
```

Comment:
minor: the extra newlines are unnecessary:

```suggestion
    appendChoice(values: string[], number: number = this._tabstop++): SnippetString {
        const value = values.map(s => s.replace(/\$|}|\\|,/g, '\\$&')).join(',');
        this.value += `\$\{${number}|${value}|\}`;
        return this;
    }
```